lib/kafka_producer.js

```diff
@@ -55,6 +56,19 @@ function KafkaProducer(options, callback) {
         } else {
             self.statsd = false;
         }
+        if ('batching' in options) {
+            self.batching = options.batching;
+        } else {
+            self.batching = false;
+        }
+
+        if (self.batching) {
+            self.maxBatchSizeBytes = options.maxBatchSizeBytes || 100000; // default 100kb buffer / match batch message per a topic
+            self.topicToBatchQueue = {}; // map of topic name to BatchMessage
+            self.flushCycleSecs = options.flushCycleSecs || 1; // flush a topic's batch message every second
+            setInterval(self.flushEntireCache, self.flushCycleSecs * 1000); // eslint-disable-line
```

Comment:
That's not how `setInterval` works. You have to do

```js
setInterval(flushEntireCache, n);

function flushEntireCache() {
  self.flushEntireCache();
}
```

Otherwise the value of `this` is incorrect.